asab/task.py

```diff
@@ -97,6 +97,25 @@ def schedule(self, *tasks):
 			self.NewTasks.put_nowait(task)
 
 
+	def schedule_threadsafe(self, *tasks):
+		"""
+		Schedule a task (or tasks) threadsafe for immediate fire-and-forget execution (e.g. compressing files).
+		Use this method to schedule task from a thread to be executed in the main thread.
+
+		Examples:
+
+		```python
+		app.TaskService.schedule_threadsafe(self._start())
+		```
+		"""
+		for task in tasks:
+
+			def do():
+				self.NewTasks.put_nowait(task)
+
+			self.App.Loop.call_soon_threadsafe(do)
```

Comment:
`self.App.Loop.call_soon_threadsafe(self.NewTasks.put_nowait, task)` ... without closure

... or do the loop in the "do()" closure